message TableRef {
    required string db_name    = 1;
    required string table_name = 2;
    optional bool use_outdated  = 3 [default = false]; //Use and outdated version of this table
};

message MetaQuery3 {
    enum MetaQueryType {
        CREATE_DB     = 1; //db_name
        DROP_DB       = 2; //db_name
        LIST_DBS      = 3;

        CREATE_TABLE  = 4;
        DROP_TABLE    = 5;
        LIST_TABLES   = 6; //db_name
    };
    required MetaQueryType type = 1;
    optional string db_name = 2;

    message CreateTable {
        optional string datacenter  = 1;
        required TableRef table_ref = 3;
        optional string primary_key = 4 [default = "id"];
        optional int64 cache_size = 5 [default = 1073741824]; //notice that's a gigabyte right there.
    }
    optional CreateTable create_table = 3;

    optional TableRef drop_table = 4;
};
